replica/database.hh

```diff
@@ -326,6 +326,40 @@ struct cf_stats {
     uint64_t total_view_updates_on_wrong_node = 0;
 };
 
+/*!
+ * \brief capacity_unit_holder represents a mechanism for tracking the consumption of capacity units.
+ *
+ * Capacity units are simplified representations of user operations within the system.
+ * Each operation performed by a user consumes a specific number of
+ * capacity units, depending on its complexity and resource requirements.
+ *
+ * The `capacity_unit_holder` class encapsulates the concept of capacity units.
+ *
+ * This class is **protocol-agnostic**, meaning it does not depend on any specific communication protocol.
+ * Each protocol that utilizes this class will define how the capacity units are updated and managed within
+ * its context.
+ *
+ * Capacity units serve two primary purposes:
+ *
+ * 1. Billing: They provide a standardized way to quantify resource usage, enabling accurate and transparent
+ *    billing based on the capacity units consumed by a user's operations.
+ *
+ * 2. Usage Limitations: They help enforce usage policies by limiting the number of capacity units
+ *    a user can consume within a given time frame.
+ *
+ */
```

Comment:
I want to please invite @avikivity to review this patch as well.
This is a main header file in Scylla, that as you said, is not only Alternator.

Reply:
I don't think this belongs in OSS

Reply:
@nyh I understand your concern about item 2. 
I can see why we may want to extend it to other metrics. 

This PR adds specific metrics to the per-table metric.
I can think of a general implementation, just need to make sure the overhead from generalization is not too high (it's not the register, it's the update that expensive).